File: python/obolary/resource/filter.py
from typing import Optional, List, Self
from pydantic import BaseModel, Field

from datetime import datetime

class Filter(BaseModel):
    
    condition : Optional[ str ] = ''
    limit_number : Optional[ int ] = 100
    limit_on : Optional[ str ] = 'created'
    limit_after_inclusive : Optional[ datetime ] = None
    limit_before : Optional[ datetime ] = None
    page : Optional[ int ] = 0
    

File: golang/resource/filter.go
package resource

import (
	"time"
)

type Filter struct {
	Condition           map[string]interface{} `json:"condition,omitempty"`
	LimitNumber         int                    `json:"limit_number,omitempty"`
	LimitOn             string                 `json:"limit_on,omitempty"`
	LimitAfterInclusive time.Time              `json:"limit_after_inclusive,omitempty"`
	LimitBefore         time.Time              `json:"limit_before,omitempty"`
	Page                int                    `json:"page,omitempty"`
}
